File: python_track/palindrome-number.cpp
class Solution {
public:
    bool isPalindrome(int x) {
        string strX = to_string(x);
        for(int start=0,end = strX.length()-1;end>start;start++,end--)
            if(strX[start]!=strX[end])
                return false;
        return true;
    }
};

File: camp_I/week4/leetcode/palindrome-number.py
class Solution:
    def isPalindrome(self, x: int) -> bool:
        if x < 0:
            return False

        tempx = x
        x_reversed = 0
        while tempx:
            x_reversed *=10
            x_reversed += tempx % 10
            tempx //= 10

        return x == x_reversed